### ISSUE TITLE
13.3.5: pytest is failing because it uses no longer maintained `random2` module

### ISSUE THREAD

**Issue by kloczek** [joke2k/faker#1635]
* Faker version: 13.3.5
* OS: Linux x86/64

### Brief summary of the issue
Looks like pytest is failing because test suite still is using outdated `random2` module which was never updated for python 3.x).

### Steps to reproduce
I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

### Expected behavior
pytest should not fail.

### Actual behavior
Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -q
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/faker-13.3.5, configfile: setup.cfg
plugins: Faker-13.3.5
collected 1491 items / 1 error

================================================================================== ERRORS ==================================================================================
_______________________________________________________________ ERROR collecting tests/providers/test_ssn.py _______________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/faker-13.3.5/tests/providers/test_ssn.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/providers/test_ssn.py:11: in <module>
    import random2
E   ModuleNotFoundError: No module named 'random2'
============================================================================= warnings summary =============================================================================
../../BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages/faker/providers/person/fr_QC/__init__.py:10
  /home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages/faker/providers/person/fr_QC/__init__.py:10: UserWarning: fr_QC locale is deprecated. Please use fr_CA.
    warnings.warn("fr_QC locale is deprecated. Please use fr_CA.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
ERROR tests/providers/test_ssn.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================= 1 warning, 1 error in 10.95s =======================================================================
```

**Comment by kloczek** [comment]
Jus tchecked and I've alredy reported that https://github.com/joke2k/faker/issues/1454.
Ticket has been closed without fixing the issue 😞

**Comment by fcurella** [comment]
@kloczek would adding `random2` to `tests_require` in `setup.py` fix your issue?

**Comment by kloczek** [comment]
I'm not using setu.py. As I wrote on top of the ticket I'm using pep517 build procedure.
Using already setup py is marked in `setuptools` as deprecated. You should move to pyproject.toml + setup.cfg.
And no .. hardcoding anywhere build dependencies to use long time not maintained module which was never updated for python 3.x is not good idea.

**Comment by fcurella** [comment]
If you have the time to work on it, I'll be more than happy to move to `pyproject.toml` and `setup.fcg`

**Comment by kloczek** [comment]
Issue is that I still have long queue packages to update/do from scratch 😞 .. however you can always count on me to report every possible small issue which I'm able to spot on packaging or test any even preliminary patch in my build env 😃 (best as PR or downloadable over URL patches because it is very easy to plug into my build procedures).
Sorry ..
If it is still unresolved just please keep that thicket opened.
In my build procedures I'm ignoring that file so that issue is not urgent for me 😄, and I fully understand that other people may have more important things to do.

**Comment by fcurella** [comment]
I've created https://github.com/joke2k/faker/issues/1636 to keep track of the efforts re: pakaging

**Comment by kloczek** [comment]
Any progress? 🤔

**Comment by kloczek** [comment]
Gentle ping ..

**Comment by stefan6419846** [comment]
Having a quick look at this, it seems like it is related to the seeds leading to a different behavior for *random* and *random2*. Taking `tests.providers.test_ssn.TestEtEE.test_ssn` as an example, using the following code will generate `46808220410` with *random* and `44010044206` with *random2*, although using the same seed:

```python3
        self.fake.random = random.Random()
        self.fake.seed_instance(0)
        print(self.fake.ssn())

        self.fake.random = random2.Random()
        self.fake.seed_instance(0)
        print(self.fake.ssn())
```

We either have to replace the target value or find the correct seed for *random* to generate the same strings. As I am not familiar enough with the SSN providers, fixing this myself might lead to wrong results. By the way: When we are already seeding the generator, shouldn't we be able to match the exact result instead?

The relevant changes have been #920 by @fcurella and #962 by @darrylwhiting.

**Comment by stefan6419846** [comment]
To add on this: I had another look at `tests.providers.test_ssn.TestEnUS.test_invalid_ssn`. Except for the second generated number, I have been able to brute-force the corresponding seeds. When checking this, I discovered that the documented numbers in the comment there are indeed not always correct - the first and the last one are correct, the middle one appears to be completely wrong and the remaining two have a last digit which is off by one.

Nevertheless, as the generated seeds are quite large, I suggest to adapt the tests for the numbers generated with *random* instead of *random2* - maybe doing exact checks for the numbers as well due to using fixed seeds anyway. If there already is some change there, dropping the *freezegun* decorators inside this test file might be valid as well.
